### DISCUSSION TITLE
Python module does not show up in nav when using auto code reference recipe

### DISCUSSION THREAD

**Discussion by cdwilson** [mkdocstrings/mkdocstrings#686] [Q&A] [answered]
I tried to follow the recipe in https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages for a minimal test python package. It looks like the page is correctly being generated from the package's `__init__.py` docstrings, but I'm just not seeing the Python module listed in the left nav (instead the "Code Reference" section is a link to the package's generated `index.md` page):

<img width="1496" alt="image" src="https://github.com/user-attachments/assets/9a30a43b-5822-4982-883a-285ae31d1f69">

Here's a minimal example which can be used to reproduce the example shown in the screenshot above:

```
git clone -b cdwilson/issues/auto-gen-api-doc-pkg-init-only git@github.com:cdwilson/sandbox-mkdocs-python.git
cd sandbox-mkdocs-python/
python3 -m venv .venv
source .venv/bin/activate
python3 -m pip install hatch
hatch run doc:serve
```

I'm new to MkDocs (and mkdocstrings) and I'm not quite sure if I'm configuring something incorrectly. I expected to see something like this in the left nav for this minimal example:

```
Code Reference <- plain section, not a link
sandbox-mkdocs-python <- link to the page generated from the package __init__.py
```

Any ideas why the package name is not shown in the left nav?

**Comment by cdwilson** [comment]
If I simply add an empty module `mymod.py` into the package, things start to work as expected (the package name shows up in the nav):

<img width="820" alt="image" src="https://github.com/user-attachments/assets/aebd7862-a2b4-4bb4-8892-1ec8519b2ecd">

<img width="1496" alt="image" src="https://github.com/user-attachments/assets/66f9c2dd-e3da-4f40-97e9-aa9a3d890db3">

**Comment by pawamoy** [comment]
To be honest, I'm not sure how to explain this behavior. I suppose it's how mkdocs-section-index / Material for MkDocs work. If there's a single section with no subpages, it gets merged into the navigation entry above :shrug: 

Summary file (literate-nav) with only the init module:

```md
* [sandbox_mkdocs_python](sandbox_mkdocs_python/index.md)
```

...contrary to:

```md
* [sandbox_mkdocs_python](sandbox_mkdocs_python/index.md)
    * [mod](sandbox_mkdocs_python/mod.md)
```

**Reply by cdwilson** [reply]
Just a quick follow up comment:

> I suppose it's how mkdocs-section-index / Material for MkDocs work.

When you mentioned `mkdocs-section-index`, I realized that I wasn't actually using it (I was using the built-in `navigation.indexes` feature instead which is supposed to do the same thing).

I changed the example above to use `section-index` instead of `navigation.indexes` and I get different behavior.

When I use `section-index`, the package actually shows up in the nav:

<img width="1496" alt="image" src="https://github.com/user-attachments/assets/ae740e39-c4ce-4bf9-af65-0acbf6ada448">

I thought that just using `section-index` might be the fix for this, but it doesn't appear to work correctly on the "Get Started" page... (where the top-level `index.md` is nested under the "Get Started" section in the nav):

<img width="1496" alt="image" src="https://github.com/user-attachments/assets/4be0d0e1-ca56-4eaf-ad7c-6bfb7786223d">

I'm not exactly sure what's causing this difference in behavior, but this might be an incompatibility between `literate-nav` and `navigation.indexes`.

**Reply by pawamoy** [reply]
IME mkdocs-section-index generally works better than navigation.indexes. It's more flexible 🤷

**Reply by cdwilson** [reply]
@pawamoy thanks for looking at this. Since this doesn't appear to be related to the recipe (or mkdocstrings in general), I created a new issue https://github.com/oprypin/mkdocs-literate-nav/issues/35 to see if maybe a solution can be found there.

**Reply by pawamoy** [reply]
You're welcome, sorry I cannot help much! Asking around is probably the best thing to do :+1:

**Comment by cdwilson** [comment, marked as answer]
FYI, I found a fix for the issues above by switching from `literate-nav` to [awesome-pages](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin).

Changes to the recipe:

```sh
pip install mkdocs-awesome-pages-plugin
```

```yaml
# mkdocs.yml

plugins:
  - ...
  - gen-files: # https://github.com/oprypin/mkdocs-gen-files
      scripts:
        - scripts/gen_ref_pages.py
  - awesome-pages # https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin
```

```yaml
# docs/.pages

nav:
  - Code Reference: reference
```

```python
# gen_ref_pages.py

"""Generate the code reference pages and navigation."""

from pathlib import Path

import mkdocs_gen_files

nav = mkdocs_gen_files.Nav()

root = Path(__file__).parent.parent
src = root / "src"

for path in sorted(src.rglob("*.py")):
    module_path = path.relative_to(src).with_suffix("")
    doc_path = path.relative_to(src).with_suffix(".md")
    full_doc_path = Path("reference", doc_path)

    parts = tuple(module_path.parts)

    if parts[-1] == "__init__":
        parts = parts[:-1]
        doc_path = doc_path.with_name("index.md")
        full_doc_path = full_doc_path.with_name("index.md")
        with mkdocs_gen_files.open(full_doc_path.with_name(".pages"), "w") as fd:
            fd.write(f"title: {parts[-1]}")
    elif parts[-1] == "__main__":
        continue

    nav[parts] = doc_path.as_posix()

    with mkdocs_gen_files.open(full_doc_path, "w") as fd:
        ident = ".".join(parts)
        fd.write(f"---\ntitle: {parts[-1]}\n---\n\n::: {ident}")

    mkdocs_gen_files.set_edit_path(full_doc_path, path.relative_to(root))

```

If you want to try it out, here's the same example from above but converted to using `awesome-pages`:

```python
git clone -b cdwilson/example/awesome-pages git@github.com:cdwilson/sandbox-mkdocs-python.git
cd sandbox-mkdocs-python/
python3 -m venv .venv
source .venv/bin/activate
python3 -m pip install hatch
hatch run doc:serve
```
